flake.nix

```diff
@@ -187,29 +187,39 @@
           # Customize source filtering as Nickel uses non-standard-Rust files like `*.lalrpop`.
           src = filterNickelSrc craneLib.filterCargoSources;
 
-          # Args passed to all `cargo` invocations by Crane.
-          cargoExtraArgs = "--frozen --offline --workspace";
-        in
-        rec {
+          # set of cargo args common to all builds
+          cargoBuildExtraArgs = "--frozen --offline";
+
           # Build *just* the cargo dependencies, so we can reuse all of that work (e.g. via cachix) when running in CI
           cargoArtifacts = craneLib.buildDepsOnly {
-            inherit
-              src
-              cargoExtraArgs;
+            inherit src;
+            cargoExtraArgs = "${cargoBuildExtraArgs} --workspace";
+            buildInputs = [ pkgs.python3 ];
```

Comment:
Please add a comment explaining why we need to add `python3`

Reply:
```suggestion
            # pyo3 needs a Python interpreter in the build environment
            # https://pyo3.rs/v0.17.3/building_and_distribution#configuring-the-python-version
            buildInputs = [ pkgs.python3 ];
```